src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -16,18 +32,24 @@ export const TodoFilter = () => (
         type="text"
         className="input"
         placeholder="Search..."
+        name="query"
+        value={query}
+        onChange={(event) => handleInput(event.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
       <span className="icon is-right" style={{ pointerEvents: 'all' }}>
```

Comment:
The button inside the span element does not have a label, which can lead to accessibility issues. Consider adding a visually hidden label or use the 'aria-label' attribute